HyperVExtension/src/HyperVExtension/Services/VMGalleryService.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Security.Cryptography;
+using System.Text.Json;
+using HyperVExtension.Models.VMGalleryJsonToClasses;
+using HyperVExtension.Providers;
+using Windows.Storage;
+
+namespace HyperVExtension.Services;
+
+/// <summary>
+/// Service that performs operations specific to the VM gallery.
+/// </summary>
+public sealed class VMGalleryService : IVMGalleryService
+{
+    private const string ComponentName = "VMGalleryService";
+
+    private readonly IDownloaderService _downloaderService;
+
+    private readonly JsonSerializerOptions _jsonOptions = new()
+    {
+        PropertyNameCaseInsensitive = true,
+        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+        TypeInfoResolver = JsonSourceGenerationContext.Default,
+        AllowTrailingCommas = true,
+    };
+
+    public VMGalleryService(IDownloaderService downloaderService)
+    {
+        _downloaderService = downloaderService;
+    }
+
+    private static readonly Uri VmGalleryUrl = new("https://go.microsoft.com/fwlink/?linkid=851584");
+
+    private VMGalleryImageList _imageList = new();
+
+    /// <inheritdoc cref="IVMGalleryService.GetGalleryImagesAsync"/>
+    public async Task<VMGalleryImageList> GetGalleryImagesAsync()
+    {
+        // If we have already downloaded the image list before, return it.
+        if (_imageList.Images.Count > 0)
+        {
+            return _imageList;
+        }
+
+        var emptyList = new VMGalleryImageList();
+        try
+        {
+            var cancellationTokenSource = new CancellationTokenSource();
+
+            // This should be quick as the file is around 28.0 KB, so a 5 minute timeout should be ok on the worst network.
+            cancellationTokenSource.CancelAfter(TimeSpan.FromMinutes(5));
```

Comment:
On 56k (dial-up) this should take 4 seconds.  A wait time of 5 minutes is huge.  I suggest limiting the call to 30 seconds.

Reply:
there could be other reasons other than bandwidth that could affect the timing, e.g lost of internet connection. I think i'll leave it as is for now. This shouldn't affect anything major